salt/modules/mount.py

```diff
@@ -1422,3 +1569,287 @@ def delete_mount_cache(real_name):
+    if isinstance(opts, list):
+        opts = ','.join(opts)
+
+    # preserve arguments for updating
+    entry_args = {
+        'name': name,
+        'dev': device.replace('\\ ', '\\040'),
+        'vfstype': vfstype,
+        'opts': opts,
+        'mount': mount,
+    }
+
+    view_lines = []
+    ret = None
+
+    if 'AIX' not in __grains__['kernel']:
+        return ret
+
+    # Transform match_on into list--items will be checked later
+    if isinstance(match_on, list):
+        pass
+    elif not isinstance(match_on, six.string_types):
+        msg = 'match_on must be a string or list of strings'
+        raise CommandExecutionError(msg)
+    elif match_on == 'auto':
+        # Try to guess right criteria for auto....
+        # added IBM types from sys/vmount.h after btrfs
+        # NOTE: missing some special fstypes here
+        specialFSes = frozenset([
+            'none',
+            'tmpfs',
+            'sysfs',
+            'proc',
+            'fusectl',
+            'debugfs',
+            'securityfs',
+            'devtmpfs',
+            'cgroup',
+            'btrfs',
+            'cdrfs',
+            'procfs',
+            'jfs',
+            'jfs2',
+            'nfs',
+            'sfs',
+            'nfs3',
+            'cachefs',
+            'udfs',
+            'cifs',
+            'namefs',
+            'pmemfs',
+            'ahafs',
+            'nfs4',
+            'autofs',
+            'stnfs'])
+
+        if vfstype in specialFSes:
+            match_on = ['name']
+        else:
+            match_on = ['dev']
+    else:
+        match_on = [match_on]
+
+    # generate entry and criteria objects, handle invalid keys in match_on
+    entry_ip = _filesystems_entry.from_line(entry_args, kwargs)
+    try:
+        criteria = entry_ip.pick(match_on)
+
+    except KeyError:
+        filterFn = lambda key: key not in _filesystems_entry.compatibility_keys
+        invalid_keys = filter(filterFn, match_on)
+
+        msg = 'Unrecognized keys in match_on: "{0}"'.format(invalid_keys)
+        raise CommandExecutionError(msg)
+
+    # parse file, use ret to cache status
+    if not os.path.isfile(config):
+        raise CommandExecutionError('Bad config file "{0}"'.format(config))
+
+    # read in block of filesystem, block starts with '/' till empty line
+    try:
+        fsys_filedict = _filesystems(config, False)
+        for fsys_view in six.viewitems(fsys_filedict):
+            if criteria.match(fsys_view):
+                ret = 'present'
+                if entry_ip.match(fsys_view):
+                    view_lines.append(fsys_view)
+                else:
+                    ret = 'change'
+                    kv = entry_ip['name']
+                    view_lines.append((kv, entry_ip))
+            else:
+                view_lines.append(fsys_view)
+
+    except (IOError, OSError) as exc:
+        msg = 'Couldn\'t read from {0}: {1}'
+        raise CommandExecutionError(msg.format(config, exc))
```

Comment:
We don't need to assign this to a variable if all we're doing is raising an exception with it.

Reply:
Due to cut & paste, as similar usage throughout existing file.

Reply:
done